File: Algorithm/Implementation/Electronics Shop/Electronics Shop.py
#!/bin/python3

import sys

def getMoneySpent(keyboards, drives, s):
    # Complete this function
    '''
    # method 1
    priceList = [keyboardPrice + drivesPrice for keyboardPrice in keyboards \
    for drivesPrice in drives if(keyboardPrice + drivesPrice) <= s]
    if not priceList: 
        return -1
    return max(priceList)
    '''
    # method 2 
    result = -1
    for i in sorted(keyboards, reverse = True):
        for j in sorted(drives):
            if i + j > s:
                break
            result = max(result, i + j)
    return result
    
s,n,m = input().strip().split(' ')
s,n,m = [int(s),int(n),int(m)]
keyboards = list(map(int, input().strip().split(' ')))
drives = list(map(int, input().strip().split(' ')))
#  The maximum amount of money she can spend on a keyboard and USB drive, or -1 if she can't purchase both items
moneySpent = getMoneySpent(keyboards, drives, s)
print(moneySpent)

File: Algorithm/Implementation/Electronics Shop/Electronics Shop.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

int getMoneySpent(int* keyboards, int* drives, int s, int n, int m){
    // Complete this function
    int i,j;
    for(i = 0; i< n - 1; i++){
        int big_index = i;
        for(j = i + 1; j < n; j++){
            if(keyboards[big_index] < keyboards[j])
                big_index = j;
        }
        int temp = keyboards[i];
        keyboards[i] = keyboards[big_index];
        keyboards[big_index] = temp;
    }
    for(i = 0; i< m - 1; i++){
        int small_index = i;
        for(j = i + 1; j < m; j++){
            if(drives[small_index] > drives[j])
                small_index = j;
        }
        int temp = drives[i];
        drives[i] = drives[small_index];
        drives[small_index] = temp;
    }
    int result = -1;
    for(i = 0,j = 0; i < n; i++){ 
        for(; j < m; j++){
            if( keyboards[i] + drives[j] > s)
                break;
            if( keyboards[i] + drives[j] > result)
                result = keyboards[i] + drives[j];
        }
    }
    return result;
}

int main() {
    int s; 
    int n; 
    int m; 
    scanf("%d %d %d", &s, &n, &m);
    int *keyboards = malloc(sizeof(int) * n);
    for(int keyboards_i = 0; keyboards_i < n; keyboards_i++){
       scanf("%d",&keyboards[keyboards_i]);
    }
    int *drives = malloc(sizeof(int) * m);
    for(int drives_i = 0; drives_i < m; drives_i++){
       scanf("%d",&drives[drives_i]);
    }
    //  The maximum amount of money she can spend on a keyboard and USB drive, or -1 if she can't purchase both items
    int moneySpent = getMoneySpent(keyboards, drives, s, n, m);
    printf("%d\n", moneySpent);
    return 0;
}
